src/override/malloc.cc

```diff
@@ -4,34 +4,78 @@
 
 using namespace snmalloc;
 
+#ifndef SNMALLOC_EXPORT
+#  define SNMALLOC_EXPORT
+#endif
+
 #ifndef SNMALLOC_NAME_MANGLE
 #  define SNMALLOC_NAME_MANGLE(a) a
 #endif
 
+namespace
+{
+  /**
+   * RAII wrapper around an `Alloc`.  This class gets an allocator from the
+   * global pool and wraps it so that `Alloc` methods can be called
+   * directly via the `->` operator on this class.  When this object is
+   * destroyed, it returns the allocator to the global pool.
+   */
+  struct slow_allocator
```

Comment:
I wonder if this class should be in a header file as it seems useful in more scenarios.